enigma-principal/app/src/epoch_u/epoch_provider.rs

```diff
@@ -30,64 +30,60 @@ use std::mem::replace;
 pub struct EpochStateManager {
     pub epoch_state_list: Mutex<Vec<EpochState>>,
     pub cap: usize,
+    pub dir_path: PathBuf,
 }
 
-impl EpochStateManager {
-    pub fn new(cap: usize) -> Result<Self, Error> {
-        let epoch_state_val = Self::read_from_file(cap)?;
-        let epoch_state_list = Mutex::new(epoch_state_val.clone());
-        Ok(Self { epoch_state_list, cap })
+fn get_file_path(mut path: PathBuf) -> Result<PathBuf, Error> {
```

Comment:
Can the path be a reference here? The argument is cloned in every call to this function, maybe it can be avoided.